front/src/modules/trainschedule/components/Timetable/PacedTrain/hooks/useOccurrenceActions.ts

```diff
@@ -147,10 +149,68 @@ const useOccurrenceActions = ({
     [pacedTrain, occurrences, selectedTrainId]
   );
 
+  /**
+   * Resets an occurrence exceptions.
+   * If it is an indexed occurrence exception, it is totally removed.
+   * If it is an added exception, every change groups are removed except the start time.
+   */
+  const resetOccurrenceExceptions = useCallback(
+    (occurrenceId: OccurrenceId) => {
+      const occurrenceToUpdateException = findExceptionWithOccurrenceId(
+        pacedTrain.exceptions,
+        occurrenceId
+      );
+
+      if (!occurrenceToUpdateException) {
+        throw new Error('Cannot reset an occurrence which was not an exception');
+      }
+
+      let updatedExceptions = [...pacedTrain.exceptions];
```

Comment:
nit: do not initialize updatedExceptions right now and use `pacedTrain.exceptions` below
```suggestion
      let updatedExceptions: PacedTrainException[];
```

Reply:
Done